debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnection.java

```diff
@@ -64,7 +64,7 @@
      * @param fieldReader binary or text protocol based readers
      */
     public MySqlConnection(MySqlConnectionConfiguration connectionConfig, MysqlFieldReader fieldReader) {
-        super(connectionConfig.config(), connectionConfig.factory(), QUOTED_CHARACTER, QUOTED_CHARACTER);
+        super(connectionConfig.jdbcConfig(), connectionConfig.factory(), QUOTED_CHARACTER, QUOTED_CHARACTER);
```

Comment:
That's interesting. There's a dedicated `Configuration` sub-type, `JdbcConfiguration`. `jdbcConfig` should be of this type, and the super constructor should expect a `JdbcConfiguration` as the first parameter, so as to avoid this kind of mistake.

Reply:
Reverse the config() method, and add originalConfig() method to avoid naming issue.